src/pymor/reductors/aaa.py

```diff
@@ -0,0 +1,427 @@
+            if err <= rel_tol:
+                break
+
+            greedy_idx = np.unravel_index(err_mat.argmax(), err_mat.shape)
+            for i in range(num_vars):
+                if greedy_idx[i] not in self.itpl_part[i] and len(self.itpl_part[i]) < max_itpl[i]:
+                    self.itpl_part[i].append(greedy_idx[i])
+
+                    # perform double interpolation step to enforce real state-space representation
+                    if i == 0 and self.conjugate and np.imag(svs[i][greedy_idx[i]]) != 0:
+                        conj_sample = np.conj(svs[i][greedy_idx[i]])
+                        conj_idx = np.where(svs[0] == conj_sample)[0]
+                        self.itpl_part[i].append(conj_idx[0])
+
+            # solve LS problem
+            L = full_nd_loewner(samples, svs, self.itpl_part)
+
+            _, S, V = spla.svd(L, lapack_driver='gesvd')
+            VH = V.T.conj()
+            coefs = VH[:, -1:]
+
+            # post-processing for non-minimal interpolants
+            d_nsp = np.sum(S/S[0] < self.nsp_tol)
+            if d_nsp > 1:
+                if self.post_process:
+                    self.logger.info('Non-minimal order interpolant computed. Starting post-processing.')
+                    pp_coefs, pp_itpl_part = _post_processing(samples, svs, self.itpl_part, d_nsp, self.L_rk_tol)
+                    if pp_coefs is not None:
+                        coefs, self.itpl_part = pp_coefs, pp_itpl_part
+                    else:
+                        self.logger.warning('Post-processing failed. Consider reducing "L_rk_tol".')
+                else:
+                    self.logger.warning('Non-minimal order interpolant computed.')
+
+            # update barycentric form
+            itpl_samples = samples[np.ix_(*self.itpl_part)]
+            itpl_samples = np.reshape(itpl_samples, -1)
+            itpl_nodes = [sv[lp] for sv, lp in zip(svs, self.itpl_part)]
+            bary_func = np.vectorize(make_bary_func(itpl_nodes, itpl_samples, coefs))
+
+            if self.post_process and d_nsp >= 1:
+                self.logger.info('Converged due to non-trivial null space of Loewner matrix after post-processing.')
+                break
+
+        # in MIMO case construct barycentric form based on matrix/vector samples
+        if dim_input != 1 or dim_output != 1:
+            itpl_samples = samples_orig[np.ix_(*self.itpl_part)]
+            itpl_samples = np.reshape(itpl_samples, (-1, dim_output, dim_input))
+
+        bary_func = make_bary_func(itpl_nodes, itpl_samples, coefs)
+
+        if self.num_vars > 1:
+            return TransferFunction(dim_input, dim_output,
+                                    lambda s, mu: bary_func(s, *(mu[p] for p in self.parameters)),
+                                    parameters=self.parameters)
+        else:
+            return TransferFunction(dim_input, dim_output, lambda s: bary_func(s))
+
+
+def nd_loewner(samples, svs, itpl_part):
```

Comment:
It seems like this function (and those below) could be member methods of `pAAAReductor`. At least, this would remove the need for function parameters like `samples` and `svs`. Would this make sense? Or would these functions be used outside of p-AAA?

Reply:
I think `_post_processing` should be a method, but the other functions can be used in other contexts. E.g., once this PR is merged implementing parametric Loewner would be quite easy and it would use all the Loewner constructions used here as well.